kong/plugins/oauth2/access.lua

```diff
@@ -643,6 +648,16 @@ local function issue_token(conf)
           end
         end
 
+        if not response_params[ERROR] and conf.global_credentials then
+          local expected_identifier = get_plugin_identifier(conf)
+          if expected_identifier ~= auth_code.plugin_identifier then
```

Comment:
As we're adding `kong.plugin.get_id()`, please remove the `expected_identifier` variable and change the condition to `auth_code.plugin_id ~= kong.plugin.get_id()`.